tests/post/steps/test_liveness.py

```diff
@@ -34,58 +35,76 @@ def check_resource_list(host, resource, namespace):
 
 
 @then(parsers.parse(
-    "we can exec '{command}' in the "
-    "'{pod}' pod in the '{namespace}' namespace"))
-def check_exec(host, command, pod, namespace):
+    "we can exec '{command}' in the pod labeled '{label}' "
+    "in the '{namespace}' namespace"))
+def check_exec(host, command, label, namespace):
+    candidates = _get_pods(host, label, namespace)
+    try:
+        pod, = candidates
```

Comment:
I'm still not fond of the trailing comma trick since it's hard to spot.

Could be rewritten:

```
    if len(pos) != 1:
        pytest.fail(
            "Expected one (and only one) pod with label {l}, found {f}".format(
                l=label, f=len(candidates)
            )
        )
    pod = candidates[0]
```
The code is even shorter by one line while being more explicit.

Reply:
Tracked [here](https://github.com/scality/metalk8s/issues/781#issuecomment-476565028).